content/posts/introducing-delouse.md

```diff
@@ -0,0 +1,158 @@
+from our own binary), an endpoint to crash the process and generate a
+`coredump`, an endpoint to request a Rust stacktrace, and finally an endpoint
+to request for a `tokio` stacktrace. Let’s try out the `tokio` stacktrace
+endpoint and take a look at our heartbeat task above.
+
+```
+$ curl http://localhost:7132/stacktrace/tokio | jq -r .stacktrace
+Tokio Task ID 0:
+╼ <hyper::server::shutdown::Graceful<I,S,F,E> as core::future::future::Future>::poll at /home/paultag/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hyper-0.14.28/src/server/shutdown.rs:78:32
+    (omitted due to length)
+
+Tokio Task ID 1:
+╼ <dropshot::server::HttpServer<C> as core::future::future::Future>::poll at /home/paultag/.cargo/registry/src/index.crates.io-6f17d22bba15001f/dropshot-0.10.0/src/server.rs:747:9
+    (omitted due to length)
+
+Tokio Task ID 2:
+╼ heartbeat::main::{{closure}}::{{closure}} at /home/paultag/dev/kittycad/delouse/examples/heartbeat.rs:9:67
+  └╼ <tokio::time::sleep::Sleep as core::future::future::Future>::poll at /home/paultag/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/time/sleep.rs:448:22
+     └╼ tokio::time::sleep::Sleep::poll_elapsed at /home/paultag/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/time/sleep.rs:404:16
+
+
+Tokio Task ID 3:
+╼ <hyper::server::server::new_svc::NewSvcTask<I,N,S,E,W> as core::future::future::Future>::poll at /home/paultag/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hyper-0.14.28/src/server/server.rs:777:36
+    (omitted due to length)
+
+Tokio Task ID 4:
+╼ dropshot::server::http_request_handle::{{closure}}::{{closure}} at /home/paultag/.cargo/registry/src/index.crates.io-6f17d22bba15001f/dropshot-0.10.0/src/server.rs:918:69
+    (omitted due to length)
+```
+
+We can see that we’ve got 4 tasks running, including our heartbeat task running
+in Task 2. We can see it’s currently sleeping, and on what line it’s sleeping
+(`heartbeat.rs:9`). Running this against our Design API quickly narrowed the
+problem down to our WebRTC handlers reading from the same shared messaging
+channel -- allowing us to push a small fix to explicitly close the shared
+resource, which resulted in spinning down any tasks spawned from a closed
+WebSocket connection. Our metrics looked completely fixed the next day – and
+we’ve not seen any issues in the following weeks.
+
+The `delouse` endpoints will gracefully degrade when system support is not
+possible or configured. For example, the `tokio` stacktrace support is only
+implemented upstream for Linux on `aarch64`, `x86` and `x86_64`. In addition to
+OS support, the `tokio` stacktrace endpoint requires that the `tokio_unstable`
+and `tokio_taskdump` config options are set -- which means adding some flags to
+your project’s `.cargo/config.toml` -- the `README` has some `cargo`-cultable
+examples. More than anything, though, the `delouse` hooks are designed to be
```

Comment:
What does it mean for something to be `cargo`-cultable? I've only heard the cultural reference, but is cargo-cult a Rust thing? I'm vaguely remembering something called cargo-cult but I forget what it does

Reply:
It was the cultural reference - but it's also confusing and in the way of the point -- trimmed this too, thank you!